collector/nvd/src/server.rs

```diff
@@ -134,13 +128,18 @@ pub fn config(cfg: &mut web::ServiceConfig, auth: Option<Arc<Authenticator>>) {
     .service(SwaggerUi::new("/swagger-ui/{_:.*}").url("/openapi.json", ApiDoc::openapi()));
 }
 
-pub async fn register_with_collectorist(state: &SharedState, advertise: Option<Url>) {
+pub async fn register_with_collectorist(state: &AppState, advertise: Option<Url>) -> anyhow::Result<()> {
+    log::info!("Starting collectorist loop - advertise: {advertise:?}");
+
     loop {
         if let Some(addr) = *state.addr.read().await {
             if !state.connected.load(Ordering::Relaxed) {
-                let url = advertise
-                    .clone()
-                    .unwrap_or_else(|| Url::parse(&format!("http://{}:{}/api/v1/", addr.ip(), addr.port())).unwrap());
+                let url = match &advertise {
+                    Some(url) => url.clone(),
+                    None => {
+                        Url::parse(&format!("http://{addr}/api/v1/")).context("Failed to build advertisement URL")?
```

Comment:
Should this be https or is this branch usually not taken in that case?

Reply:
It should be TLS, once the client side is migrated to TLS too. Right now, I am focusing on "allowed the server side to serve TLS"